### ISSUE TITLE
`bun add` crashes Bun when `"workspaces"` contains a non-existent glob pattern

### ISSUE THREAD

**Issue by jack-weilage** [oven-sh/bun#10220]
### What version of Bun is running?

1.1.3+2615dc742

### What platform is your computer?

Linux 6.5.0-26-generic x86_64 x86_64

### What steps can reproduce the bug?

```bash
echo '{ "workspaces": ["doesnt-exist/*"] }' > package.json
bun add -d @types/bun # Can be any package
```

### What is the expected behavior?

Bun should successfully add the package and warn that the "doesnt-exist/*" directory doesn't exist, similar to the behavior of `bun install`:

```bash
> bun i
bun install v1.1.3 (2615dc74)
1 | { "workspaces": ["doesnt-exist/*"] }
     ^
warn: workspaces directory prefix not found "doesnt-exist/"
   at /tmp/test/package.json:1:2
```

### What do you see instead?

```bash
> bun add -d @types/bun
bun add v1.1.3 (2615dc74)
2 |   "workspaces": ["doesnt-exist/*"],
      ^
warn: workspaces directory prefix not found "doesnt-exist/"
   at /tmp/test/package.json:2:3

error: InstallFailed

----- bun meta -----
Bun v1.1.3 (2615dc74) Linux x64 #26~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Tue Mar 12 10:22:43 UTC 2
AddCommand: 
Elapsed: 5ms | User: 4ms | Sys: 8ms
RSS: 35.66MB | Peak: 24.90MB | Commit: 35.66MB | Faults: 0
----- bun meta -----

0   0x58f22365835b
1   ???
2   ???
3   ???
4   ???
5   ???
6   ???

Search GitHub issues https://bun.sh/issues or ask for #help in https://bun.sh/discord
```

### Additional information

Related to https://github.com/oven-sh/bun/issues/2456, but `bun add` crashes Bun entirely here so IMO this is a separate issue.

**Comment by jack-weilage** [comment]
As of Bun v1.1.5 the result of running the `add` command is 
```sh
> bun add -d @types/bun
bun add v1.1.5 (b257a309)
2 |   "workspaces": ["doesnt-exist/*"],
      ^
warn: workspaces directory prefix not found "doesnt-exist/"
   at /tmp/test/package.json:2:3
error: An internal error ocurred (InstallFailed)
```

**Comment by dylan-conway** [comment]
fixed in #10426, and included in `v1.1.8`
